src/sentry_database.c

```diff
@@ -121,7 +149,40 @@ sentry__process_old_runs(const sentry_options_t *options)
     while ((run_dir = sentry__pathiter_next(db_iter)) != NULL) {
         // skip over other files such as the saved consent or the last_crash
         // timestamp
-        if (!sentry__path_is_dir(run_dir)) {
+        if (!sentry__path_is_dir(run_dir)
+            || !sentry__path_ends_with(run_dir, ".run")) {
+            continue;
+        }
+
+        sentry_stringbuilder_t sb;
+        sentry__stringbuilder_init(&sb);
+        const sentry_pathchar_t *filename = sentry__path_filename(run_dir);
+#ifdef SENTRY_PLATFORM_WINDOWS
+        char *filename_c = sentry__string_from_wstr(filename);
+        if (filename_c) {
+            sentry__stringbuilder_append(&sb, filename_c);
+            sentry_free(filename_c);
+        }
+#else
+        sentry__stringbuilder_append(&sb, filename);
+#endif
+        // `<db>/<uuid>.run.lock`
+        sentry__stringbuilder_append(&sb, ".lock");
+        char *lockfile_s = sentry__stringbuilder_into_string(&sb);
+        if (!lockfile_s) {
+            continue;
+        }
+        sentry_path_t *lockfile
+            = sentry__path_join_str(options->database_path, lockfile_s);
```

Comment:
_nit_: This entire part where you build the path might be nice to extract into a function for readability. The code itself looks quite good to me, but it's a bit overwhelming in the context of this function with all the `continue`.

Reply:
BTH, all this string manipulation, also with the fact that we have normal, and wstr variants makes this a great pain in the rear. Lets land it like this and rethink it another time…